sqlmesh/core/model/common.py

```diff
@@ -19,7 +19,8 @@ def parse_model_name(name: str) -> t.Tuple[t.Optional[str], t.Optional[str], str
     Returns:
         A tuple consisting of catalog, schema, table name.
     """
-    return split_num_words(name, ".", 3)  # type: ignore
+    table = exp.to_table(name)
+    return table.catalog or None, table.db or None, table.name
```

Comment:
why do you need or None

Reply:
do we use this anywhere else?

Reply:
Removed: https://github.com/TobikoData/sqlmesh/pull/1130/commits/79db74c91162a57081296be3f2f2707bf90b4fb2